src/consumer/consumer.js

```diff
@@ -371,6 +372,41 @@ module.exports = class ConsumerGroup {
     })
   }
 
+  async assignPartitions() {
+    const memberAssignment = this.topics.map(topic => {
+      const partitionMetadata = this.cluster.findTopicPartitionMetadata(topic)
+      return {
+        topic,
+        partitions: partitionMetadata.map(metadata => metadata.partitionId),
+      }
+    })
+    this.subscriptionState.assign(memberAssignment)
+    this.partitionsPerSubscribedTopic = this.generatePartitionsPerSubscribedTopic()
+
+    const offsetManagerMemberAssignment = memberAssignment.reduce(
+      (partitionsByTopic, { topic, partitions }) => ({
+        ...partitionsByTopic,
+        [topic]: partitions,
```

Comment:
isnt this anti parttern /.
https://www.richsnapp.com/article/2019/06-09-reduce-spread-anti-pattern